app/src/ble.c

```diff
@@ -222,6 +222,23 @@ int zmk_ble_clear_bonds() {
     return 0;
 };
 
+int zmk_ble_clear_all_bonds() {
+    LOG_DBG("zmk_ble_clear_all_bonds()");
+
+    // Unpair all profiles
+    for (uint8_t i = 0; i < ZMK_BLE_PROFILE_COUNT; i++) {
```

Comment:
```suggestion
    for (int i = 0; i < ZMK_BLE_PROFILE_COUNT; i++) {
```
Using integers of small types as loop counters is generally a bad idea. In this specific case, it doesn't matter because ZMK_BLE_PROFILE_COUNT is a compile-time constant which *shouldn't* be bigger than 255, so the compiler will optimize this to the same code no matter what type is used here, but see https://stackoverflow.com/questions/34891181/for-array-loop-use-int-or-uint8-t

Reply:
I'm quite happy to change this if it better matches the project style (and have done so), but the concern I have is that this isn't indexing an arbitrary array as described in your link, it's calling a function with a `uint8_t` argument type. It doesn't ring quite as true to me to iterate with a wider type that will need a narrowing cast to pass to the called function. Probably just a nit-pick, so it's your call.